### PR TITLE
Add script to compare 2 TimeTests runs

### PR DESCRIPTION
**Pull request by vurusovs**
### Tickets:
 - *-48864

### CODE REVIEW

File: tests/time_tests/scripts/compare_runs.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    # Report will contain data with order defined in `required_fields` and `required_metrics`
+    required_fields = [
+        # "results" should be excluded because it will be handled automatically
+        "model_name", "precision", "framework", "device", "timetest", "platform", "os"
+    ]
+    required_metrics = [
+        "first_inference_latency"
+    ]
+    # `Ops` is a template applied for every metric defined in `required_metrics`
+    ops = OrderedDict([
```
Comment (vurusovs) [inline tests/time_tests/scripts/compare_runs.py:69]: think about how to compare results including stdev

File: tests/time_tests/scripts/compare_runs.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+                             f' data using reference source.',
+                        choices=DB_COLLECTIONS)
+    parser.add_argument('--out_file', dest='output_file', type=Path,
+                        help='Path to a file to save results.')
+
+    args = parser.parse_args()
+    return args
+
+
+def _parse_platform(platform):
```
Comment (vurusovs) [inline tests/time_tests/scripts/compare_runs.py:174]: docstring

File: tests/time_tests/scripts/compare_runs.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        return platform[beg:end]
+    elif "win_" in platform:
+        beg = platform.find("win_") + len("win_")
+        end = platform.find("&&", beg)
+        return platform[beg:end]
+    else:
+        raise Exception("Failed to parse platform name")
+
+
+def _update_timetests_records(records: list):
```
Comment (vurusovs) [inline tests/time_tests/scripts/compare_runs.py:187]: docstring

File: tests/time_tests/scripts/compare_runs.py [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    orig_data = pandas.DataFrame(comparison_data)
+    data = orig_data.dropna()
+    
+    devices = data["device"].unique()
+    for device in devices:
+        frame = data[data["device"] == device]
+        for field, comp_op in comparison_ops.items():
+            status, msg = comp_op(frame.loc[:, field])
+            if not status:
+                log.error('Comparison for field="%s" for device="%s" failed: %s', field, device, msg)
```
Comment (vurusovs) [inline tests/time_tests/scripts/compare_runs.py:134]: add information about comparison status and metrics in report (HTML or any other report)

**Comment by asomsiko** [comment]
data analytics should be through an internal web reporter tool. Could you clarify how the proposed tool will be used and why we need a tool additionally to the web reporter?

**Review by asomsiko — CHANGES_REQUESTED** [review]
it looks a duplicate to what is available in web reporter

**Comment by vurusovs** [comment]
Closed as not required for now

**Pull request closed without merging**